packages/CopyInput/src/CopyInput.js

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Input from "@paprika/input";
+import Popover from "@paprika/popover";
+import Button from "@paprika/button";
+import useI18n from "@paprika/l10n/lib/useI18n";
+import extractChildrenProps from "@paprika/helpers/lib/extractChildrenProps";
+import CopyIcon from "@paprika/icon/lib/ArrowDown"; // TODO: Replace with actual icon when ready
+import CopyInputInputPropsCollector from "./CopyInputInputPropsCollector";
+import * as sc from "./CopyInput.styles";
+
+function CopyInput(props) {
+  const { defaultValue, children, ...moreProps } = props;
+  const I18n = useI18n();
+  const extendedInputProps = extractChildrenProps(children, CopyInputInputPropsCollector);
+
+  const [value, setValue] = React.useState(defaultValue);
+  const [isClickedTooltipOpen, setIsClickedTooltipOpen] = React.useState(false);
+
+  function handleButtonClick() {
+    navigator.clipboard.writeText(value);
+    setIsClickedTooltipOpen(true);
+    setTimeout(() => setIsClickedTooltipOpen(false), 5000);
+  }
+
+  return (
+    <sc.CopyInput data-pka-anchor="copy-input" {...moreProps}>
+      <Input defaultValue={defaultValue} onChange={e => setValue(e.target.value)} {...extendedInputProps} />
```

Comment:
What if a user of this component wants to use the `value`. E.g.

```js
const [controlledValue, setControlledValue] = React.useState(defaultValue);

<CopyInput defaultValue="Copy me">
  <CopyInput.Input isReadOnly value={controlledValue} onChange={e => setControlledValue(e.target.value)} />
</CopyInput.Input>
```

In that case user would have to override the default `onChange`, hence, the `value` inside of the `CopyInput` will not be updated. And that will defeat the purpose of the component.

Reply:
I was going to add this as my feedback as well, this component seems that should be a controlled component.

Reply:
thanks for catching that, i'll fix it